include/boost/url/impl/const_string.hpp

```diff
@@ -92,8 +45,27 @@ const_string(
     std::size_t n,
     Allocator const& a,
     char*& dest)
-    : p_(construct(n, a, dest))
+    : refs_(nullptr),
+      alloc_(a)
 {
+    if(n == 0)
+    {
+        static_cast<string_view&>(*this) = { nullptr, n };
```

Comment:
or
```
*this = string_view();
```
?

Reply:
```
error: no viable overloaded '='
```

It might also imply we want to call the string_view constructor, which would lose the allocator.